### PR TITLE
Sequence game to display correct/incorrect on completion of the sequence selection

### PR DESCRIPTION
**Pull request by KaranChimple**

### CODE REVIEW

File: lib/quiz/sequence.dart [hunk 1]
```diff
@@ -21,8 +21,10 @@ class SequenceQuiz extends StatefulWidget {
 class SequenceQuizState extends State<SequenceQuiz> {
   String ans;
   int score = 0;
-  bool clicked = false;
-  var choice = [];
+  var clicked = [];
```
Comment (talapadi) [inline lib/quiz/sequence.dart]: Specify the type

File: lib/quiz/sequence.dart [hunk 2]
```diff
@@ -21,8 +21,10 @@ class SequenceQuiz extends StatefulWidget {
...
+  var choice = [], clickedChoices = [];
```
Comment (talapadi) [inline lib/quiz/sequence.dart]: Specify the type

File: lib/quiz/sequence.dart [hunk 3]
```diff
@@ -31,93 +33,88 @@ class SequenceQuizState extends State<SequenceQuiz> {
...
+    // Array for storing choices from input
     for (var i = 0; i < widget.input['order'].length; i++) {
       choice.add(widget.input['order'][i]);
     }
     // choice = choice.map((a) => widget.input['order'][a]).toList(growable: false);
     ans = widget.input['image'];
     print("Choices at initializtion -$choice");
+    
+    // Array to track the choices clicked
+    clicked = choice.map((e) => "false").toList(growable: false);
```
Comment (talapadi) [inline lib/quiz/sequence.dart]: Do you want the String 'false' or bool 'false'? It is dangerous to rely on strings when all you need is a boolean value

File: lib/quiz/sequence.dart [hunk 4]
```diff
@@ -31,93 +33,88 @@ class SequenceQuizState extends State<SequenceQuiz> {
...
-            } else {
-              score = 0;
+          // changing value of clicked button to true when button is clicked
+          setState(() {
+                      clicked[k] = "true";
+                      clickedChoices.add(text); // storing the sequence in which the choices are clicked
+                      count++; // counter to check if the sequence is completed
+                    });
+
+            if(count == 4){
```
Comment (talapadi) [inline lib/quiz/sequence.dart]: Are we hardcoding the number of choices to 4?

File: lib/quiz/sequence.dart [hunk 5]
```diff
@@ -31,93 +33,88 @@ class SequenceQuizState extends State<SequenceQuiz> {
...
-      tableRows.add(new Column(
-        children: cells,
-        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-      ));
-    }
 
-    return new Scaffold(
-      backgroundColor: Colors.grey,
-      body: new Column(
+    return new LayoutBuilder(
```
Comment (talapadi) [inline lib/quiz/sequence.dart]: Why do you need a LayoutBuilder when you are not using its constraints?

File: lib/quiz/sequence.dart [hunk 6]
```diff
@@ -31,93 +33,88 @@ class SequenceQuizState extends State<SequenceQuiz> {
   }
 
   void _initboard() {
```
Comment (talapadi) [inline lib/quiz/sequence.dart]: Let all names be camelCase

File: lib/quiz/sequence.dart (hunk 3 above)
Comment (KaranChimple) [reply lib/quiz/sequence.dart]: I'm not just relying on a Boolean value. I'm later setting the value of clicked array to "completed" so as to inform that all the buttons have been clicked and its time to display options in green/red

File: lib/db/dao/quiz_progress_dao.dart [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import 'dart:async';
+import 'package:maui/app_database.dart';
+import 'package:maui/db/entity/quiz_progress.dart';
+import 'package:maui/db/entity/user.dart';
+import 'package:maui/db/entity/topic.dart';
+import 'package:maui/db/entity/quiz.dart';
+import 'package:sqflite/sqflite.dart';
+
+class QuizProgressDao {
+  Future<QuizProgress> getQuizprogress(String id, {Database db}) async {
```
Comment (talapadi) [inline lib/db/dao/quiz_progress_dao.dart]: getQuizProgress

File: lib/db/dao/quiz_progress_dao.dart [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      return QuizProgress.fromMap(maps.first);
+    }
+    return null;
+  }
+
+  Future<List<QuizProgress>> getQuizProgressByTopicId(String topicId,
+      {Database db}) async {
+    db = db ?? await new AppDatabase().getDb();
+    List<Map> maps =
+        await db.query('${QuizProgress.table} a,${Topic.table} t', columns: [
```
Comment (talapadi) [inline lib/db/dao/quiz_progress_dao.dart]: Dont need the join with the Topic table

File: lib/db/dao/quiz_progress_dao.dart [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      topicId
+    ]);
+    return maps.map((el) => new QuizProgress.fromMap(el)).toList();
+  }
+
+  Future<List<QuizProgress>> getQuizProgressByUserId(String userId,
+      {Database db}) async {
+    db = db ?? await new AppDatabase().getDb();
+    List<Map> maps =
+        await db.query('${QuizProgress.table} a,${User.table} u', columns: [
```
Comment (talapadi) [inline lib/db/dao/quiz_progress_dao.dart]: No need for join with User table

File: lib/db/dao/quiz_progress_dao.dart [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+      userId
+    ]);
+    return maps.map((el) => new QuizProgress.fromMap(el)).toList();
+  }
+
+  Future<List<QuizProgress>> getQuizProgressByQuizId(String quizId,
+      {Database db}) async {
+    db = db ?? await new AppDatabase().getDb();
+    List<Map> maps =
+        await db.query('${QuizProgress.table} a,${Quiz.table} q', columns: [
```
Comment (talapadi) [inline lib/db/dao/quiz_progress_dao.dart]: No need for join with Quiz table

**Pull request merged**